sawmill-core/src/main/java/io/logz/sawmill/processors/GeoIpProcessor.java

```diff
@@ -20,40 +20,71 @@
 import io.logz.sawmill.exceptions.SawmillException;
 import io.logz.sawmill.utilities.JsonUtils;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
 
 import javax.inject.Inject;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.InetAddress;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.EnumSet;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.zip.GZIPInputStream;
 
 import static com.google.common.base.Preconditions.checkState;
 import static io.logz.sawmill.processors.GeoIpProcessor.Property.ALL_PROPERTIES;
 import static io.logz.sawmill.processors.GeoIpProcessor.Property.LOCATION;
 import static java.util.Collections.EMPTY_LIST;
 import static java.util.Objects.requireNonNull;
 
+@SuppressWarnings("UnstableApiUsage")
 @ProcessorProvider(type = "geoIp", factory = GeoIpProcessor.Factory.class)
 public class GeoIpProcessor implements Processor {
 
+    private static final String TAR_GZ_SUFFIX = ".tar.gz";
+
     private static DatabaseReader databaseReader;
 
     @VisibleForTesting
     static void loadDatabaseReader(String location) {
```

Comment:
Add a test verifying tar.gz database loading (along the .mmdb have a packed tar.gz containing it)